From 2bfbfbf2f39b8e10b702e182263accb9c0b3293a Mon Sep 17 00:00:00 2001
From: HeYue <yue.he@intel.com>
Date: Mon, 12 Jun 2023 16:55:33 +0800
Subject: [PATCH] Revert "audio: exit(1) if audio backend failed to be found or
 initialized"

Revert commit: 0f957c53c84d655f2e99677d407cf2bbe1832de4
It causes the qemu launch fail. Qemu exit if audio
backend failed to be found or initialized.

Signed-off-by: HeYue <yue.he@intel.com>
---
 audio/audio.c | 15 ++++-----------
 audio/audio.h |  2 +-
 softmmu/vl.c  |  4 +---
 3 files changed, 6 insertions(+), 15 deletions(-)

diff --git a/audio/audio.c b/audio/audio.c
index 065602ce1..3c2b4f77f 100644
--- a/audio/audio.c
+++ b/audio/audio.c
@@ -1771,6 +1771,8 @@ static AudioState *audio_init(Audiodev *dev, const char *name)
         atexit_registered = true;
     }
 
+    QTAILQ_INSERT_TAIL(&audio_states, s, list);
+
     s->ts = timer_new_ns(QEMU_CLOCK_VIRTUAL, audio_timer, s);
 
     s->nb_hw_voices_out = audio_get_pdo_out(dev)->voices;
@@ -1795,10 +1797,6 @@ static AudioState *audio_init(Audiodev *dev, const char *name)
         } else {
             dolog ("Unknown audio driver `%s'\n", drvname);
         }
-        if (!done) {
-            free_audio_state(s);
-            return NULL;
-        }
     } else {
         for (i = 0; audio_prio_list[i]; i++) {
             AudiodevListEntry *e = audiodev_find(&head, audio_prio_list[i]);
@@ -1836,7 +1834,6 @@ static AudioState *audio_init(Audiodev *dev, const char *name)
                "(Audio can continue looping even after stopping the VM)\n");
     }
 
-    QTAILQ_INSERT_TAIL(&audio_states, s, list);
     QLIST_INIT (&s->card_head);
     vmstate_register (NULL, 0, &vmstate_audio, s);
     return s;
@@ -2169,17 +2166,13 @@ void audio_define(Audiodev *dev)
     QSIMPLEQ_INSERT_TAIL(&audiodevs, e, next);
 }
 
-bool audio_init_audiodevs(void)
+void audio_init_audiodevs(void)
 {
     AudiodevListEntry *e;
 
     QSIMPLEQ_FOREACH(e, &audiodevs, next) {
-        if (!audio_init(e->dev, NULL)) {
-            return false;
-        }
+	    audio_init(e->dev, NULL);
     }
-
-    return true;
 }
 
 audsettings audiodev_to_audsettings(AudiodevPerDirectionOptions *pdo)
diff --git a/audio/audio.h b/audio/audio.h
index 01bdc567f..20b03d514 100644
--- a/audio/audio.h
+++ b/audio/audio.h
@@ -170,7 +170,7 @@ void audio_sample_from_uint64(void *samples, int pos,
 
 void audio_define(Audiodev *audio);
 void audio_parse_option(const char *opt);
-bool audio_init_audiodevs(void);
+void audio_init_audiodevs(void);
 void audio_help(void);
 void audio_legacy_help(void);
 
diff --git a/softmmu/vl.c b/softmmu/vl.c
index 5115221ef..e47865cd2 100644
--- a/softmmu/vl.c
+++ b/softmmu/vl.c
@@ -1909,9 +1909,7 @@ static void qemu_create_early_backends(void)
      * setting machine properties, so they can be referred to.
      */
     configure_blockdev(&bdo_queue, machine_class, snapshot);
-    if (!audio_init_audiodevs()) {
-        exit(1);
-    }
+    audio_init_audiodevs();
 }
 
 
-- 
2.25.1

